Description: add change_password call
 Add change_password call to Authen::Krb5
 .
 libauthen-krb5-perl (1.9-3build1) precise; urgency=low
 .
Author: Hanno Hecker <hanno.hecker@zalando.de>

---
--- libauthen-krb5-perl-1.9.orig/Krb5.xs
+++ libauthen-krb5-perl-1.9/Krb5.xs
@@ -648,6 +648,33 @@ krb5_recvauth(auth_context,fh,version,se
    sv_setref_pv(ST(0),"Authen::Krb5::Ticket",(void*)ticket);
    XSRETURN(1);
 
+void
+krb5_change_password(creds, newpw)
+    Authen::Krb5::Creds creds
+    char *newpw
+
+    PREINIT:
+    int result_code;
+    krb5_data result_code_string;
+    krb5_data result_string;
+    char msg[4096];
+
+    PPCODE:
+    err = krb5_change_password(context, creds, newpw, &result_code,
+        &result_code_string, &result_string);
+    if (err) XSRETURN_UNDEF;
+    if (result_code) {
+        if (result_code_string.length + result_string.length + 5 >
+            sizeof(msg)) {
+            sprintf(msg, "error too long\n");
+        } else {
+            sprintf(msg, "%.*s%s%.*s", result_code_string.length, result_code_string.data,
+                        result_string.length?": ":"",
+                        (int) result_string.length,result_string.data ? result_string.data : "");
+        }
+        croak(msg);
+    }
+    XSRETURN_YES;
 
 MODULE = Authen::Krb5  PACKAGE = Authen::Krb5::Principal
 
--- libauthen-krb5-perl-1.9.orig/Krb5.pm
+++ libauthen-krb5-perl-1.9/Krb5.pm
@@ -311,6 +311,12 @@ replay cache.  Returns the replay cache
 
 Returns a Authen::Krb5::Rcache object using the replay cache name 'name.'
 
+=item change_password(creds, newpw)
+
+Change the password for the principal contained in the creds. 'creds' must be
+obtained via the get_init_creds_keytab or get_init_creds_password call. B<NOTE>: croak()s
+on error (i.e. the underlying 'result_code' is non zero). Returns true on success.
+
 =back
 
 =head2 CLASSES & METHODS

